R/secondary_weights.R

```diff
@@ -39,49 +39,99 @@ secondary_weights <- function(secondary_raster, grid = era5_grid, extent = "full
   # Create ERA raster from input raster
   clim_raster <- raster::raster(grid) # only reads the first band
 
-  ## Raster alignment: make sure clim_raster is -180 to 180 longitude
+  ## Raster alignment: make sure clim_raster is in same coordinate system as secondary
   ## -----------------------------------------------
 
   message(crayon::yellow('Checking for raster alignment'))
 
-  poly_xmin <- -180
-  poly_xmax <- 180
-  rast_xmin <- raster::extent(clim_raster)@xmin
-  rast_xmax <- raster::extent(clim_raster)@xmax
+  ## secondary raster
+  s_rast_xmax <- raster::extent(secondary_raster)@xmax
+  s_rast_res <- raster::xres(secondary_raster)
 
-  # Rotate raster if initial longitudes don't align
-  if(!dplyr::near(poly_xmax, rast_xmax, tol=1.01)) {
+  s_rast_coord <- if(s_rast_xmax > 180 + s_rast_res / 2) {
+    "climate (0-360)"
+  } else {
+    "standard (-180 to 180)"}
 
-    message(crayon::yellow('Adjusting raster longitude from',
-                           round(rast_xmin,0), '-', round(rast_xmax,0),
-                           'to', round(poly_xmin,0), '-', round(poly_xmax,0)))
+  message(crayon::yellow('Secondary raster coordinate system:', s_rast_coord))
 
-    clim_raster <- raster::rotate(clim_raster)
+  ## climate raster
+  c_rast_xmax <- raster::extent(clim_raster)@xmax
+  c_rast_res <- raster::xres(clim_raster)
 
-  }
+  c_rast_coord <- if(c_rast_xmax > 180 + c_rast_res / 2) {
+    "climate (0-360)"
+  } else {
+    "standard (-180 to 180)"}
+
+  message(crayon::yellow('Grid coordinate system:', c_rast_coord))
+
+  ## check if coordinate systems match, if no shift raster in 0-360 format
+  if(s_rast_coord != c_rast_coord) {
+
+    message(crayon::yellow('Coordinate systems do not match. Adjusting raster longitude
+                           to standard coordiantes between -180 and 180 degrees.'))
+
```

Comment:
Maybe we can get rid of this message? It seems like the two messages in 91 and 108 are repetitive with this one but those specify which coordinate system is being changed so it seems like that gives more relevant info than this one.